### ISSUE TITLE
InputNumber: 6.2 RC2 exception when decode invoked on empty submitted value

### ISSUE THREAD

**Issue by martin654** [primefaces/primefaces#3330]
## 1) Environment
- PrimeFaces version: 6.2 RC2
- Does it work on the newest released PrimeFaces version? Version? yes, 6.1.13
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source)
- Application server + version: any
- Affected browsers: any

## 2) Expected behavior
p:inputNumber processes empty submitted value as usual. 

## 3) Actual behavior
NumberFormatException is thrown.
```
java.lang.NumberFormatException: null
	at java.math.BigDecimal.<init>(BigDecimal.java:596)
	at java.math.BigDecimal.<init>(BigDecimal.java:383)
	at java.math.BigDecimal.<init>(BigDecimal.java:806)
	at org.primefaces.component.inputnumber.InputNumberRenderer.decode(InputNumberRenderer.java:72)
	at javax.faces.component.UIComponentBase.decode(UIComponentBase.java:832)
```

## 4) Steps to reproduce
Open showcase page for inputNumber, leave "Empty value (sign)" empty and click new "crash" button.

## 5) Sample XHTML
Add button to inputNumber.xhtml page:

`<p:commandButton process="@form" update="@form" value="crash" />`

Bug was introduced in issue https://github.com/primefaces/primefaces/issues/3239

**Comment by tandraschko** [comment]
@cnsgithub Could you check that, too?

**Comment by tandraschko** [comment]
Also if the exception isn't nice, we should think about the case is valid.
I think emptyString or blank is valid and we need to add a trim - however a string value like "crash" shoudln't be valid.

**Comment by cnsgithub** [comment]
I was able to reproduce this issue.

As there is no natural reset functionality in the component this state can only appear when using with uninitialized wrapped values (e.g. `Double `or `Integer `being `null`) and `required="false"` (default) or when manually resetting the value as shown above. Though, I confirm, that's a valid use case.

I fixed this in the previous commit. Exceptions for invalid inputs will also be caught and wrapped by `FacesException`.

However, care must be taken when value is bound to primitive (`double`, `int`) instead of wrapper (`Double`, `Integer`). Then the converter's `getAsObject` will return `null `for an empty string input and somehow instead of throwing a `NullPointerException` and rejecting the `null` value it's converted further to `0` or` 0.0` which may be out of the [`minValue`, `maxValue`] range. Where exactly does this magic in the update model phase happen, in PrimeFaces or the JSF implementation?
